java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/deploy/DeployCommand.java

```diff
@@ -123,6 +120,9 @@ public class DeployCommand implements Runnable {
     @CommandLine.Option(names = "--git-disable-ssl-verification")
     boolean gitDisableSSLVerification;
 
+    @CommandLine.Option(names = "--git-reuse-repository")
```

Comment:
Why is this nessesary? The repo is already re-used if it exists?

Reply:
Yes exactly. If the dependencybuild is created directly there is an option in the CLI to reuse the existing scm to push back changes i.e. if we're using a midstream instead of a pristine upstream. This is then cascaded through to here so that not only the prebuild-source task can push back to midstream but so can deploy (which currently runs the pre/post build scripts).